### ISSUE TITLE
Readonly object can be used as parameter for function expecting mutable object

### ISSUE THREAD

**Issue by xaviergonz** [Microsoft/TypeScript#23176]
**TypeScript Version:**  2.8.1

**Search Terms:** mutate readonly immutable mutable

**Code**

```ts
interface Mutable {
  x: number;
}

interface Immutable {
  readonly x: number;
}

function mutate(mut: Mutable) {
  mut.x = 5;
}

function immutate(imm: Immutable) {
  // imm.x = 5; // this does not compile as expected
  mutate(imm); // but this does
}
```

**Expected behavior:** There should be a compiler error inside immutate saying that 'imm' cannot be used as a parameter to mutate since 'imm' is readonly while 'mut' is not

**Actual behavior:** Compiles successfully

**Playground Link:**  https://codesandbox.io/s/vnp9k2z32y

**Related Issues:**

**Comment by xaviergonz** [comment]
Another example

```ts
interface Mutable {
  x: number;
}

interface Immutable {
  readonly x: number;
}

const imm: Immutable = {x: 3};
// imm.x = 5; // not allowed, good :D
const mut: Mutable = imm;
mut.x = 5; // allowed :-/

// vs

const roA: ReadonlyArray<number> = [5];
// const writeA: number[] = roA; // not allowed, good :D
```

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/11180
